packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -33,7 +33,7 @@ class FlutterErrorDetails {
     this.stackFilter,
     this.informationCollector,
     this.silent: false
-  });
+  }) : assert(exception != null);
```

Comment:
I don't think we should assert here. If this ever fires, it's going to be confusing as heck. In practice the code handles null fine. It's just never supposed to be null because that indicates another error somewhere (like when people show the error message for ID 0 and it says "error: no error").